pkg/filesystem/path/builder_test.go

```diff
@@ -31,13 +37,65 @@ func TestBuilder(t *testing.T) {
 			"/hello/../world/foo",
 		} {
 			t.Run(p, func(t *testing.T) {
+				// Unix Parser
 				builder1, scopewalker1 := path.EmptyBuilder.Join(path.VoidScopeWalker)
 				require.NoError(t, path.Resolve(path.MustNewUNIXParser(p), scopewalker1))
 				require.Equal(t, p, builder1.GetUNIXString())
 
 				builder2, scopewalker2 := path.EmptyBuilder.Join(path.VoidScopeWalker)
 				require.NoError(t, path.Resolve(builder1, scopewalker2))
 				require.Equal(t, p, builder2.GetUNIXString())
+
+				// Windows Parser, compare Windows and UNIX string identity.
+				windowsStyle := strings.Replace(p, "/", "\\", -1)
+				builder3, scopewalker3 := path.EmptyBuilder.Join(path.VoidScopeWalker)
+				require.NoError(t, path.Resolve(path.MustNewWindowsParser(p), scopewalker3))
+				require.Equal(t, windowsStyle, getWindowsString(t, builder3))
+				require.Equal(t, p, builder3.GetUNIXString())
+
+				builder4, scopewalker4 := path.EmptyBuilder.Join(path.VoidScopeWalker)
+				require.NoError(t, path.Resolve(builder3, scopewalker4))
+				require.Equal(t, windowsStyle, getWindowsString(t, builder4))
+				require.Equal(t, p, builder4.GetUNIXString())
+			})
+		}
+
+		for _, p := range []string{
+			"C:\\",
+			"C:\\hello\\",
+			"C:\\hello\\..",
+			"C:\\hello\\..\\world",
+			"C:\\hello\\..\\world\\",
+			"C:\\hello\\..\\world\\foo",
+		} {
+			t.Run(p, func(t *testing.T) {
+				// Drive letter paths. These are printed as qualified absolute
+				// paths by the Windows string formatter, and as bare absolute
+				// UNIX paths without a drive letter.
+				noDriveletter := p[2:]
+				unixStyle := strings.Replace(noDriveletter, "\\", "/", -1)
+				driveletter := "C:" + noDriveletter
```

Comment:
https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

I would prefer it if we could refrain from string concatenation and `strings.Replace()` as part of these tests. Maybe it's best if this test purely focused on the interaction between `NewWindowsParser()` and `GetWindowsString()`, because as the name implies this is about the identity relationship between them.

If we want to test the interaction between `NewWindowsString()` and `GetUNIXString()`, that should likely be provided as part of a separate test? (Maybe also rename "Identity" to "UNIXIdentity", so that this can become a separate "WindowsIdentity"?)

Reply:
Sure thing!